catroid/src/main/java/org/catrobat/catroid/formulaeditor/common/FormulaElementOperations.kt

```diff
@@ -0,0 +1,308 @@
+ */
+package org.catrobat.catroid.formulaeditor.common
+
+import android.content.res.Resources
+import com.badlogic.gdx.math.Rectangle
+import org.catrobat.catroid.common.Constants
+import org.catrobat.catroid.common.LookData
+import org.catrobat.catroid.content.GroupSprite
+import org.catrobat.catroid.content.Look
+import org.catrobat.catroid.content.Project
+import org.catrobat.catroid.content.Scene
+import org.catrobat.catroid.content.Sprite
+import org.catrobat.catroid.formulaeditor.FormulaElement
+import org.catrobat.catroid.formulaeditor.SensorHandler
+import org.catrobat.catroid.formulaeditor.Sensors
+import org.catrobat.catroid.formulaeditor.UserList
+import org.catrobat.catroid.formulaeditor.UserVariable
+import org.catrobat.catroid.nfc.NfcHandler
+import org.catrobat.catroid.sensing.CollisionDetection
+import org.catrobat.catroid.stage.StageActivity
+import org.catrobat.catroid.stage.StageListener
+import org.catrobat.catroid.utils.NumberFormats
+import org.catrobat.catroid.utils.TouchUtil
+import java.lang.Double.valueOf
+import kotlin.math.round
+
+object FormulaElementOperations {
+    @JvmStatic
+    fun getLookLayerIndex(sprite: Sprite?, look: Look, spriteList: List<Sprite?>): Double {
+        val lookZIndex = look.zIndex
+        return when {
+            lookZIndex == 0 -> 0.0
+            lookZIndex < 0 -> spriteList.indexOf(sprite).toDouble()
+            else -> lookZIndex.toDouble() - Constants.Z_INDEX_NUMBER_VIRTUAL_LAYERS
+        }
+    }
+
+    @JvmStatic
+    fun equalsDoubleIEEE754(left: Double, right: Double) =
+        left.isNaN() && right.isNaN() || !left.isNaN() && !right.isNaN() && left >= right && left <= right
+
+    @JvmStatic
+    fun interpretOperatorEqual(left: Any, right: Any): Boolean {
+        val leftString = left.toString()
+        val rightString = right.toString()
+        return try {
+            equalsDoubleIEEE754(leftString.toDouble(), rightString.toDouble())
+        } catch (_: NumberFormatException) {
+            leftString == rightString
+        }
+    }
+
+    @JvmStatic
+    fun tryInterpretDoubleValue(obj: Any): Double {
+        return when (obj) {
+            is String -> try {
+                valueOf(obj)
+            } catch (_: NumberFormatException) {
+                Double.NaN
+            }
+            else -> obj as Double
+        }
+    }
+
+    @JvmStatic
+    fun normalizeDegeneratedDoubleValues(value: Any?): Any {
+        return when (value) {
+            is String,
+            is Char -> value
+            is Double -> when (value) {
+                Double.NEGATIVE_INFINITY -> -Double.MAX_VALUE
+                Double.POSITIVE_INFINITY -> Double.MAX_VALUE
+                else -> value
+            }
+            else -> 0.0
+        }
+    }
+
+    @JvmStatic
+    fun isInteger(value: Double) = value.isFinite() && value == round(value)
+
+    @JvmStatic
+    fun tryGetLookBackgroundNumber(lookData: LookData?, lookDataList: List<LookData>) =
```

Comment:
This searches general for the index of a look, not only Background. Please rename it